kotlin-analysis-api/src/test/kotlin/com/google/devtools/ksp/impl/test/AbstractKSPAATest.kt

```diff
@@ -94,8 +105,22 @@ abstract class AbstractKSPAATest : AbstractKSPTest(FrontendKinds.FIR) {
             kotlinCoreEnvironment.projectEnvironment.environment.jarFileSystem as CoreJarFileSystem
         )
 
-        val resolver = ResolverAAImpl(ktFiles)
-        testProcessor.process(resolver)
+        val testRoot = mainModule.testRoot
+
+        val kspOptions = KspOptions.Builder().apply {
+            if (!mainModule.javaFiles.isEmpty()) {
+                javaSourceRoots.add(mainModule.javaDir)
+            }
+            classOutputDir = File(testRoot, "kspTest/classes/main")
+            javaOutputDir = File(testRoot, "kspTest/src/main/java")
+            kotlinOutputDir = File(testRoot, "kspTest/src/main/kotlin")
+            resourceOutputDir = File(testRoot, "kspTest/src/main/resources")
+            projectBaseDir = testRoot
+            cachesDir = File(testRoot, "kspTest/kspCaches")
+            kspOutputDir = File(testRoot, "kspTest")
+        }.build()
+        val ksp = KotlinSymbolProcessing(compilerConfiguration, kspOptions, CommandLineKSPLogger(), testProcessor)
```

Comment:
`KotlinCoreEnvironment` is created twice.

Reply:
TODO: check KotlinCoreEnvironment.createForTests

Reply:
This relates to a flag in CoreApplicationEnvironment and is consumed widely, where the comment says this will make the ApplicationEnvironment dispose after use, let's stick to test only KotlinCoreEnvironment for now.